### PR TITLE
Better documentation on desktop app development

### PR DESCRIPTION
**Pull request by timsneath**
_Description of what this PR is changing or adding, and why:_
Expand documentation for desktop app development with more information about unique challenges on each platform, links to relevant platform packages, and guidance on adopting the look and feel on each platform. Also splits this out into subpages. Also attempts to use the Oxford comma sufficiently to pass @sfshaza2's high standards :) 

_Issues fixed by this PR (if any):_

## Presubmit checklist
- [X] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [X This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [X] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/desktop/macos.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+for compatibility with macOS 10.15+.
+This requires enabling the Hardened Runtime option.
+Once you have enabled it, you need a valid signing
+certificate in order to build.
+
+By default, the entitlements file allows JIT for
+debug builds but, as with App Sandbox, you may
+need to manage other entitlements.
+If you have both App Sandbox and Hardened
+Runtime enabled, you may need to add multiple
```
Comment (sfshaza2) [inline src/desktop/macos.md:157]:
```suggestion
debug builds but, as with App Sandbox, you might
need to manage other entitlements.
If you have both App Sandbox and Hardened
Runtime enabled, you might need to add multiple
```

File: src/desktop/windows.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+[packages support Windows]: {{site.pub}}/packages?q=platform%3Awindows
+[`url_launcher`]: {{site.pub-pkg}}/url_launcher
+[`shared_preferences`]: {{site.pub-pkg}}/shared_preferences
+[`file_selector`]: {{site.pub-pkg}}/file_selector
+[`path_provider`]: {{site.pub-pkg}}/path_provider
+
+## Supporting Windows UI guidelines
+
+While you can use any visual style or theme you choose, including Material, some
+app authors may wish to build an app that matches the conventions of Microsoft's
```
Comment (sfshaza2) [inline src/desktop/windows.md:58]:
```suggestion
app authors might wish to build an app that matches the conventions of Microsoft's
```

File: src/desktop/windows.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+[Fluent design system][]. The [fluent_ui][] package, a [Flutter Favorite][],
+provides support for visuals and common controls that are commonly found in
+modern Windows apps, including navigation views, content dialogs, flyouts, date
+pickers, and tree view widgets.
+
+In addition, Microsoft offers [fluentui_system_icons][], a package which
```
Comment (sfshaza2) [inline src/desktop/windows.md:64]:
```suggestion
In addition, Microsoft offers [fluentui_system_icons][], a package that
```

**Review by sfshaza2 — APPROVED** [review]
lgtm. Wow, you don't give an editor much to work with, other than may->might.  :D

**Review by domesticmouse — APPROVED** [review]

**Comment by timsneath** [comment]
You're training me well, @sfshaza2 :)

**Pull request merged**